Added withdrawal features + docs for Calling an Airnode guide (#493)
api3dao/vitepress-docs: docs/guides/airnode/calling-an-airnode/index.md

> [Open in Remix<ExternalLinkImage/>](https://remix.ethereum.org/#url=https://github.com/vanshwassan/RemixContracts/blob/master/contracts/Requester.sol&optimize=false&runs=200&evmVersion=null&version=soljson-v0.8.9+commit.e5eed63a.js)

Reviewer: The new withdrawal functionality isn't included when clicking this "Open in Remix" link

Author: Fixed it with a new link

```solidity

> //SPDX-License-Identifier: MIT

Reviewer: Why delete this space character? I'm not sure if there is a perfectly agreed-upon standard, but [SPDX itself has the space in there](https://spdx.dev/ids/).

pragma solidity 0.8.9;
// [!code focus:2]
import "@api3/airnode-protocol/contracts/rrp/requesters/RrpRequesterV0.sol";
import "@openzeppelin/contracts/access/Ownable.sol";

// A Requester that will return the requested data by calling the specified Airnode.
contract Requester is RrpRequesterV0, Ownable {
    mapping(bytes32 => bool) public incomingFulfillments;
    mapping(bytes32 => int256) public fulfilledData;

> // Make sure you specify the right _rrpAddress for your chain while deploying the contract.

Reviewer: Why remove the indentation here and elsewhere? It looks a bit strange in the contract code: https://vitepress-docs--pr493-requester-with-withd-ggoohfx2.web.app/guides/airnode/calling-an-airnode/#_1-coding-the-requester-contract

Author: Fixed 👍🏻

        incomingFulfillments[requestId] = true;
    }

    function fulfill(bytes32 requestId, bytes calldata data)
        external
        onlyAirnodeRrp
    {
        require(incomingFulfillments[requestId], "No such request made");
        delete incomingFulfillments[requestId];
        int256 decodedData = abi.decode(data, (int256));
        fulfilledData[requestId] = decodedData;
    }

    // To withdraw funds from the sponsor wallet to the contract.
    function withdraw(address airnode, address sponsorWallet) external onlyOwner {

>         airnodeRrp.requestWithdrawal(
>         airnode,
>         sponsorWallet
>         );

Reviewer: nit: perhaps format as:
```suggestion
        airnodeRrp.requestWithdrawal(airnode, sponsorWallet);
```

Author: Done 👍🏻

  been encoded and contains a timestamp in addition to other response data.
  Decode it using the function decode() from the abi object.

> ### `withdraw()` method

Reviewer: Perhaps "Withdrawal" or something similar given the other h3 headings in this section are "Request Parameters" and "Response Parameters"

Author: Yeah, I agree. Fixed this too 👍🏻

::: details Coingecko's Airnode Details

```
Coingecko's Airnode Address = 0x478f9e54E5B2ee1D0A05cae1FaA3591CAA1b7091
Coingecko's Airnode XPUB = xpub6CnJr6BxgEpTBs9Aso71S5EkcRsR4MHzzKGTRA8aoyYhyvqgcacZFnx6a6quX47Vo97oACohk2isNAz68jnmDqrxQ9C1dKWMZtDN6tyVHaG

> Coingecko's Endpoint ID (`/simple/price`) = 0xfb87102cdabadf905321521ba0b3cbf74ad09c5d400ac2eccdbef8d6143e78c4

Reviewer: This needs to be updated in the code snippet below

> ![Making the Request](src/s7.png)

## 4. Withdrawing Funds from the sponsor wallet (optional)

You can withdraw funds from the sponsor wallet to your Requester contract by
calling the `withdraw()` function.

The Airnode listens for withdrawal requests and fulfills them automatically.
Therefore, the sponsor (in this case, the Requester contract itself) should be

> able to receive their funds from their `sponsorWallet` in a few minutes notice.

Reviewer: ```suggestion
able to receive their funds from their `sponsorWallet` within a few minutes.
```

Author: I've changed this part entirely. Instead of having the user call `withdrawFromContract()`, I added it under the `receive()` function to send the funds to the owner automatically. Let me know if it looks fine.

Reviewer: Oh nice idea 👍
